compiler/src/iree/compiler/Codegen/Common/MaterializeEncodingIntoPadding.cpp

```diff
@@ -291,11 +292,9 @@ struct MaterializeEncodingIntoPaddingPass final
     } else {
       IREE::GPU::TargetAttr targetAttr = getCLGPUTarget(context);
       SmallVector<NamedAttribute> items;
-      items.emplace_back(IREE::Encoding::kEncodingResolverAttrName,
-                         IREE::GPU::GPUPadLayoutAttr::get(
-                             context, /*cacheLineBytes=*/std::nullopt,
-                             /*cacheSets=*/std::nullopt));
-      items.emplace_back(kGPUTargetAttrName, targetAttr);
+      items.emplace_back(
+          IREE::Encoding::kEncodingResolverAttrName,
+          IREE::GPU::getHIPTargetEncodingLayoutAttr(targetAttr, "pad"));
       targetConfig = DictionaryAttr::get(context, items);
```

Comment:
FYI, you don't need a vector at all here. ArrayRef has a constructor that takes a single element so you only need a local variable for `NamedAttribute`.

Reply:
Thanks, I'll iterate on this in a follow-up! I finally get a proper fix in https://github.com/iree-org/iree/pull/20393, this one is going to be closed because I had wrong understanding about the encoding!